### PR TITLE
Logic fix on counting replies and creating audit log entriens.

### PR DESCRIPTION
**Pull request by mrhaekkinen**
Logic fix on counting replies and creating audit log entriens. In addition some autoformatting applied and fixed some warnings.

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt [hunk 1]
```diff
@@ -28,594 +28,613 @@ import java.time.OffsetDateTime
...
+                    replyId = reply.id!!,
+                    attachmentId = null,
+                    action = null,
+                    type = AuditLogEntryType.LIST_REPLY
+            )
+        }
+        return createOk(result, resultCount)
+    }
+
+    private fun getReplyIdList(replyIds: List<UUID>, firstResult: Int?, maxResults: Int?): List<UUID> {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:430]: Add docs

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt [hunk 2]
```diff
@@ -28,594 +28,613 @@ import java.time.OffsetDateTime
...
+        if (firstResult == null && maxResults == null) {
+            return replyIds
+        }
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:433]: Isn't this redundant? Next line checks whether firstResult is null and the following if maxResults is null

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt [hunk 3]
```diff
@@ -28,594 +28,613 @@ import java.time.OffsetDateTime
...
+
+        if (lastResult > maxResults) {
+            return replyIds.subList(firstResult, replyIds.count()-1)
+        }
+
+        if (lastResult > replyIds.count()) {
+            return emptyList()
+        }
+
+        println("sublist requested from index $firstResult to $lastResult")
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:451]: Remove println

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt [hunk 4]
```diff
@@ -28,594 +28,613 @@ import java.time.OffsetDateTime
...
+        return replyIds.subList(firstResult, lastResult)
+    }
+
+    /**
+     * Filters out replies without permission
+     *
+     * @param metaformId metaform id
+     * @param replies replies
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:459]: Fix docs

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt (hunk 1 above)
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:430]: Both privates should be located after public methods

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt [hunk 5]
```diff
@@ -704,26 +704,21 @@ class ReplyTestsIT : AbstractTest() {
...
             testBuilder.test1.replies.createSimpleReply(metaform.id, "pagination-test-3", ReplyMode.CUMULATIVE)
-
             val allReplies = testBuilder.systemAdmin.replies.listReplies(metaform.id, null, null, null, null, null, null, null, null, null, null, null)
             assertEquals(allReplies.size, 3)
             assertEquals(allReplies[0].data!!["text"], "pagination-test-1")
             assertEquals(allReplies[1].data!!["text"], "pagination-test-2")
             assertEquals(allReplies[2].data!!["text"], "pagination-test-3")
-
-            val replies1To2 = testBuilder.systemAdmin.replies.listReplies(metaform.id, null, null, null, null, null, null, null, null, 2, null, null)
+            val replies1To2 = testBuilder.systemAdmin.replies.listReplies(metaform.id, null, null, null, null, null, null, null, 0, 2, null, null)
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt:715]: This test no longer tests whether null firstResult with maxResults work, is this change intentional?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/ReplyController.kt [hunk 6]
```diff
@@ -37,9 +37,11 @@ import jakarta.enterprise.context.ApplicationScoped
 import jakarta.enterprise.event.Event
 import jakarta.inject.Inject
 
+data class ReplyIdAndResourceId(val id: UUID, val resourceId: UUID?)
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/controllers/ReplyController.kt:40]: Added docs

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/ReplyController.kt [hunk 7]
```diff
@@ -375,7 +377,7 @@ class ReplyController {
      * @param latestFirst return the latest result first according to the criteria in orderBy
      * @return replies list of replies
      */
-    fun listReplies(
+    fun listIdsAndResourceIds(
```
[the remark this answers is not in the document]
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/controllers/ReplyController.kt:380]: Docs fixed

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt (hunk 3 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:451]: Removed

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt (hunk 4 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:459]: Docs fixed

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt (hunk 1 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:430]: Moved to the down below ;)

File: src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt (hunk 2 above)
Comment (mrhaekkinen) [reply src/main/kotlin/fi/metatavu/metaform/server/rest/RepliesApi.kt:433]: Yes.

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt (hunk 5 above)
Comment (mrhaekkinen) [reply src/test/java/fi/metatavu/metaform/server/test/functional/tests/ReplyTestsIT.kt:715]: Unintentional change, reverted

**Review by villejuutila — APPROVED** [review]
